ingen/data_source/file_source.py

```diff
@@ -64,4 +64,6 @@ def format_file_path(self, source, params_map):
         path_parser = PathParser(run_date)
         if 'file_path' in source:
             source['file_path'] = path_parser.parse(source['file_path'])
+        if not source.get('file_encoding'):
```

Comment:
the function `format_file_path` is not the right place for this. We can pass `file_encoding` in the `_src` like mentioned in the other comment and then default it to utf-8 while reading in `XMLFileReader`